packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -483,30 +478,142 @@ export class PerformanceEvent {
     constructor(
         message: string,
-        props?: ITelemetryProperties,
+        props?: ITaggableTelemetryProperties,
     ) {
         super(message);
+        this.addTelemetryProperties(props);
+    }
+
+    /**
+     * Add additional properties to be logged
+     */
+    public addTelemetryProperties(props?: ITaggableTelemetryProperties) {
         Object.assign(this, props);
     }
 
-    // Return all properties
+    /**
+     * Enumerate this object's own properties to find those fit for telemetry.
+     * This protects us when someone sets a non-loggable field on this object (via 'as any')
+     */
+    private getOwnValidTelemetryProps(): ITaggableTelemetryProperties {
+        const props: ITaggableTelemetryProperties = {};
+        for (const key of Object.keys(this)) {
+            const val = this[key];
+            switch (typeof val) {
+                case "string":
+                case "number":
+                case "boolean":
+                case "undefined":
+                    props[key] = val;
+                    break;
+                default: {
+                    if (IsTaggedTelemetryPropertyValue(val)) {
+                        props[key] = val;
+                    }
+                    break;
+                }
+            }
+        }
+        return props;
+    }
+
+    /**
+     * Get all properties fit to be logged to telemetry for this error
+     */
     public getTelemetryProperties(): ITelemetryProperties {
-        const props: ITelemetryProperties = {};
-        // Could not use {...this} because it does not return properties of base class.
-        for (const key of Object.getOwnPropertyNames(this)) {
-            props[key] = this[key];
+        // Copy non-enumerable props inherited from Error
+        const props: ITelemetryProperties = {
+            stack: this.stack,
+            message: this.message,
+            name: this.name,
+        };
+
+        // Walk the object's telemetry props, respecting the tags and converting them to ITelemetryProperties
+        // Note: Once ITelemetryBaseLogger is updated to use ITaggableTelemetryProperties,
+        // this loop will go away and we can just return the taggable properties directly for the host to deal with
+        const taggableProps = this.getOwnValidTelemetryProps();
+        for (const key of Object.keys(taggableProps)) {
+            const taggableProp = taggableProps[key];
+            const { value, tag } = (typeof taggableProp === "object")
+                ? taggableProp
+                : { value: taggableProp, tag: TelemetryDataTag.None };
+            switch (tag) {
+                case TelemetryDataTag.None:
+                case TelemetryDataTag.PackageData:
+                    // For Microsoft applications, PackageData is safe for now (we don't load 3P code in 1P apps)
+                    // But this determination really belongs in the host layer
+                    props[key] = value;
```

Comment:
i think this is the wrong place to expand, see below, but when we expand i think the tag should be in the key, key_tag or tag_key

Reply:
Hmm not sure about that. Let's discuss next week.